File: schemas/podcast-platform/convex/show_tags.ts
// show_tags: tags associated with shows for discovery and categorisation.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const show_tags = defineTable({
  showId: v.id("shows"),
  tag: v.string(),
})
  .index("by_show_id_tag", ["showId", "tag"])
  .index("by_tag", ["tag"]);

File: schemas/podcast-platform/spacetimedb/show_tags.rs
// show_tags: Freeform tags applied to shows for discovery and search.
// See README.md for full design rationale.

#[spacetimedb::table(name = show_tags, public)]
pub struct ShowTag {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub show_id: String, // UUID — FK → shows.id (cascade delete)
    #[index(btree)]
    pub tag: String,
}
// Composite unique: (show_id, tag) — enforce in reducer logic
